runtime/src/append_vec.rs

```diff
@@ -174,14 +208,27 @@ impl AppendVec {
 
     #[allow(clippy::mutex_atomic)]
     pub fn set_file<P: AsRef<Path>>(&mut self, path: P) -> io::Result<()> {
-        self.path = path.as_ref().to_path_buf();
```

Comment:
Only update `self.path` in an atomic way if everything went successful [here](https://github.com/solana-labs/solana/pull/7373/files#diff-657d1e7648a9fb74cb1582a9c48beb4bR231).